src/main/java/com/conveyal/r5/analyst/GridComputer.java

```diff
@@ -180,31 +183,82 @@ public void run() throws IOException {
                 LOG.warn("Disparate direct modes and access modes are not supported in analysis mode.");
             }
-            LinkedPointSet linkedDestinationsDirect = destinations.link(network.streetLayer, accessMode);
+            LinkedPointSet linkedDestinationsAccess = destinations.link(network.streetLayer, accessMode);
 
             // Transit search, run the raptor algorithm to get times at each destination for each iteration
             LOG.info("Maximum number of rides: {}", request.request.maxRides);
             LOG.info("Maximum trip duration: {}", request.request.maxTripDurationMinutes);
 
             // first, find the access stops
             StreetRouter sr = new StreetRouter(network.streetLayer);
-            sr.distanceLimitMeters = 2000; // TODO hardwired same as traveltimesurfacecomputer
-            sr.streetMode = accessMode;
             sr.profileRequest = request.request;
-            sr.setOrigin(request.request.fromLat, request.request.fromLon);
-            sr.dominanceVariable = StreetRouter.State.RoutingVariable.DISTANCE_MILLIMETERS;
-            sr.route();
 
-            TIntIntMap reachedStops = sr.getReachedStops();
+            TIntIntMap accessTimes;
+            int offstreetTravelSpeedMillimetersPerSecond = (int) (request.request.getSpeed(accessMode) * 1000);
+            int[] nonTransitTravelTimesToDestinations;
+
+            if (request.request.accessModes.contains(LegMode.CAR_PARK)) {
+                //Currently first search from origin to P+R is hardcoded as time dominance variable for Max car time seconds
+                //Second search from P+R to stops is not actually a search we just return list of all reached stops for each found P+R.
+                // If multiple P+Rs reach the same stop, only one with shortest time is returned. Stops were searched for during graph building phase.
+                // time to stop is time from CAR streetrouter to stop + CAR PARK time + time to walk to stop based on request walk speed
+                //by default 20 CAR PARKS are found it can be changed with sr.maxVertices variable
+                sr = PointToPointQuery.findParkRidePath(request.request, sr, network.transitLayer);
+
+                if (sr == null) {
+                    // Origin not found. Return an empty access times map, as is done by the other conditions for other modes.
+                    // TODO this is ugly. we should have a way to break out of the search early (here and in other methods).
+                    accessTimes = new TIntIntHashMap();
+                } else {
+                    accessTimes = sr.getReachedStops();
+                }
+
+                // disallow non-transit access
+                // TODO should we allow non transit access with park and ride?
+                nonTransitTravelTimesToDestinations = new int[linkedDestinationsAccess.size()];
+                Arrays.fill(nonTransitTravelTimesToDestinations, RaptorWorker.UNREACHED);
```

Comment:
I did this because when you have a park and ride search I don't think we ever want to return non-transit times (because the non-transit times would be from driving, not park-and-ride; I don't think people requesting a park-and-ride journey want to see some journeys that are just driving all the way to the destination).